src/arr/base/contracts.arr

```diff
@@ -33,13 +40,17 @@ data FieldFailure:
   | missing-field(loc, field) with:
     render-reason(self, loc, from-fail-arg):
       [ED.error:
-        [ED.para: ED.text("Missing field"), ED.code(ED.text(self.field)),
-          ED.text("is required at"), draw-and-highlight(self.loc)]]
+        [ED.para: ED.text("Missing field "), ED.code(ED.text(self.field)),
```

Comment:
I would rather nix `ED.para-nospace` altogether and follow a rule of what-you-write-is-what-you-get. `ED.para-nospace` seems like a needless complication. Let's say I want to write a helper function that takes some argument and depending on its value outputs one string or another. If we keep the `ED.para-nospace` and `ED.para` distinction, I cannot trust that that function will behave how I intended; I need to specialize it for the space-injecting and no-space scenarios and remember to call the right one.